runtime/rasdump/dmpsup.c

```diff
@@ -577,7 +578,32 @@ configureDumpAgents(J9JavaVM *vm)
 		}
 		if ( IS_CONSUMABLE(j9vm_args, xdumpIndex) && !IS_CONSUMED(j9vm_args, xdumpIndex) )
 		{
-			GET_OPTION_VALUE(xdumpIndex, ':', &optionString);
+			/* Handle mapped tool dump options */
+			if (HAS_MAPPING(j9vm_args, xdumpIndex)) {
+				char *toolCursor = strstr(MAPPING_J9NAME(j9vm_args, xdumpIndex), ":tool") + 1;
+				if (NULL != toolCursor) {
+					/* The mapped option specifies the tool command to run after the equals */
+					char *optionValue = NULL;
+					GET_OPTION_VALUE(xdumpIndex, '=', &optionValue);
+					
+					/* Construct optionString by combining the J9 tool dump command with the mapped option */
+					int toolCursorLength = strlen(toolCursor);
+					int optionValueLength = strlen(optionValue);
+					optionString = (char *) j9mem_allocate_memory(toolCursorLength + optionValueLength + 1, OMRMEM_CATEGORY_VM);
+
+					if (NULL == optionString) {
+						j9tty_err_printf(PORTLIB, "Storage for mapped tool dump optionString not available, unable to process option\n");
+						return J9VMDLLMAIN_FAILED;
```

Comment:
@pdbain-ibm This is the added check for memory allocation failure. I went with `J9VMDLLMAIN_FAILED` for the error code since it was being used for a similar check in the method, but I noticed that `J9VMDLLMAIN_SILENT_EXIT_VM` and `OMR_ERROR_OUT_OF_NATIVE_MEMORY` are also used in other checks. Is there a particular error code I should use for this situation?

Reply:
Looks okay to me, thought the committers may have other opinions.
As discussed, please see if there is a test for the existing mapped option and extend that.